manual/manual/tutorials/coreexamples.etex

```diff
@@ -580,6 +580,69 @@ let fixpoint f x =
 the function "f" cannot raise a "Done" exception, which removes an
 entire class of misbehaving functions.
 
+\section{Lazy expressions}
+
+OCaml allows to defer some computation, usually expensive one, until later when 
+we need that expression evaluated. The deferred expression has type "'a Lazy.t", 
+where "'a" is the type of the evaluated expression. 
+
+We use "lazy (expr)" to delay the evaluation of some expression "expr". For 
+example, we can defer the computation of "1+1" until we need the result of that
+expression, i.e. "2". 
+
+\begin{caml_example}{toplevel}
+let lazy_two = lazy (1+1);;
```

Comment:
You could add a small side-effect to the expression, to make it clearer when the expression is evaluated. For instance,
```OCaml
let lazy_two = lazy ( print_endline "lazy_two evaluation"; 1 + 1 )
```

Reply:
Thanks, that's a good idea.